File: study/lua/src/app/scene/view/seasonCompetitive/SquadAvatar.lua
-- @Author panhoa
-- @Date 8.24.2018
-- @Role SquadAvatar

local ViewBase = require("app.ui.ViewBase")
local SquadAvatar = class("SquadAvatar", ViewBase)

function SquadAvatar:ctor()
    self._commonHeroAvatar  = nil   -- avatar
    self._heroId            = nil   -- heroId

    local resource = {
		file = Path.getCSB("SquadAvatar", "seasonCompetitive"),

	}
	SquadAvatar.super.ctor(self, resource)
end

function SquadAvatar:onCreate()
    --
end

function SquadAvatar:onEnter()
    --
end

function SquadAvatar:onExit()
    --
end

-- @Role Update avatar Info
function SquadAvatar:updateUI(heroId, limitLevel)
    --
    self._commonHeroAvatar:updateUI(heroId, "", false, limitLevel)
    self._heroId = heroId
end

-- @Role Show effect
function SquadAvatar:showAvatarEffect(bShow)
    --
    self._commonHeroAvatar:showAvatarEffect(true)
end

-- @Role Scale
function SquadAvatar:setScale(scale)
    --
    self._commonHeroAvatar:setScale(scale)
end

-- @Role Get HeroId
function SquadAvatar:getHeroId()
    --
    return self._heroId
end

-- @Role Play idle
function SquadAvatar:playAnimationNormal()
	-- body
	self._commonHeroAvatar:setAniTimeScale(1)
	self._commonHeroAvatar:setAction("idle", true)
end

-- @Role Set opacity
function SquadAvatar:setOpacity(opacity)
    return self._commonHeroAvatar:setOpacity(opacity)
end

-- @Role
function SquadAvatar:getSpine()
    return self._commonHeroAvatar
end

-- @Role trun
function SquadAvatar:turnBack(bTrue)
    self._commonHeroAvatar:turnBack(bTrue)
end

return SquadAvatar

File: study/ts/script/scene/view/seasonCompetitive/SquadAvatar.ts
const { ccclass, property } = cc._decorator;

import CommonHeroAvatar from '../../../ui/component/CommonHeroAvatar'

@ccclass
export default class SquadAvatar extends cc.Component {

    @property({ type: CommonHeroAvatar, visible: true })
    _commonHeroAvatar: CommonHeroAvatar = null;

    private _heroId;
    private _tag;

    public updateUI(heroId, limitLevel) {
        this._commonHeroAvatar.init();
        this._commonHeroAvatar.updateUI(heroId, '', false, limitLevel);
        this._commonHeroAvatar.setTouchEnabled(false);
        this._heroId = heroId;
    }

    public showAvatarEffect(bShow) {
        this._commonHeroAvatar.showAvatarEffect(true);
    }

    public setScale(scale) {
        this._commonHeroAvatar.setScale(scale);
    }

    public getHeroId() {
        return this._heroId;
    }

    public playAnimationNormal() {
        this._commonHeroAvatar.setAniTimeScale(1);
        this._commonHeroAvatar.setAction('idle', true);
    }

    public setOpacity(opacity) {
        return this._commonHeroAvatar.node.opacity = (opacity);
    }

    public getSpine() {
        return this._commonHeroAvatar;
    }

    public turnBack(bTrue) {
        this._commonHeroAvatar.turnBack(bTrue);
    }

    public setTag(tag) {
        this._tag = tag;
    }

    public getTag() {
        return this._tag;
    }
}